### PR TITLE
feat(nft-bag): wrong-network-cta

### PR DESCRIPTION
**Pull request by JFrankfurt**
https://uniswaplabs.atlassian.net/browse/WEB-1730

I'd like to revisit this and consolidate the error messaging, click handler, and disabled state logical checks into one set. The current approach feels too error-prone to me.

### CODE REVIEW

File: src/nft/components/bag/Bag.tsx [hunk 1]
```diff
@@ -54,7 +52,9 @@ const ScrollingIndicator = ({ top, show }: SeparatorProps) => (
 )
 
 const Bag = () => {
-  const { account } = useWeb3React()
+  const { account, provider } = useWeb3React()
+  const isConnected = !!provider && !!account
```
Comment (vm) [inline src/nft/components/bag/Bag.tsx:56]: i think you could check for `isActive`?

Comment (vm) [reply src/nft/components/bag/Bag.tsx:56]: ah i guess that wouldn't work since it may be the network provider. this works then

Comment (JFrankfurt) [reply src/nft/components/bag/Bag.tsx:56]: we don't do this anywhere currently, but I think we could use `Signer.isSigner(provider)`? https://docs.ethers.io/v5/api/signer/#Signer-isSigner

Comment (vm) [reply src/nft/components/bag/Bag.tsx:56]: we probably should do this lol. ill finishing reviewing now sorry got caught up

**Review by vm — APPROVED** [review]

File: src/nft/components/bag/BagFooter.tsx [hunk 2]
```diff
@@ -74,9 +74,12 @@ const ActionButton = ({ bagStatus, onClick, walletModalIsOpen, sufficientBalance
   } else if (bagStatus === BagStatus.PROCESSING_TRANSACTION) {
     disabled = true
     buttonText = <Trans>Transaction pending</Trans>
-  } else if (sufficientBalance !== false) {
+  } else if (sufficientBalance === true) {
```
Comment (vm) [inline src/nft/components/bag/BagFooter.tsx]: can you just if (sufficientBalance)

File: src/nft/components/bag/BagFooter.tsx [hunk 3]
```diff
@@ -74,9 +74,12 @@ const ActionButton = ({ bagStatus, onClick, walletModalIsOpen, sufficientBalance
...
     disabled = false
     buttonText = <Trans>Pay</Trans>
+  } else if (sufficientBalance === false) {
```
Comment (vm) [inline src/nft/components/bag/BagFooter.tsx]: `!sufficientBalance`

**Pull request merged**